src/org/zaproxy/zap/extension/requester/ExtensionRequester.java

```diff
@@ -0,0 +1,177 @@
+/*
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2013 The ZAP development team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.requester;
+
+import java.awt.Component;
+import java.io.OutputStreamWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.swing.ImageIcon;
+import javax.swing.SwingUtilities;
+
+import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.ExtensionHookMenu;
+import org.parosproxy.paros.extension.ExtensionHookView;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.httppanel.HttpPanel;
+import org.zaproxy.zap.extension.httppanel.Message;
+
+public class ExtensionRequester extends ExtensionAdaptor {
+	
+	public static final String NAME = "ExtensionRequester";
+	
+	private static final String RESOURCE = "/org/zaproxy/zap/extension/requester/resources";	
+	
+	public static final ImageIcon REQUESTER_ICON = new ImageIcon(
+			ExtensionRequester.class.getResource( RESOURCE + "/requester.png"));
+	
+	private RequesterPanel requesterPanel = null;
+	private RightClickMsgMenuRequester popupMsgMenuRequester = null;
+	
+	private RequesterParam requesterParams;
+	private RequesterOptionsPanel requesterOptionsPanel;
+	
+	public ExtensionRequester() {
+		super();		
+		initialize();
+	}
+
+	public ExtensionRequester(String name) {
+		super(name);
+	}	
+
+	@SuppressWarnings("unchecked")
+	private void initialize() {
+		this.setName(NAME);
+		this.setOrder(201);		

```

Comment:
Consider changing to other number, it clashes (logged as error) with the one from ExtensionWappalyzer.